Sources/ClientRuntime/Auth/HTTPAuthAPI/AuthScheme.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public protocol AuthScheme {
+    var schemeID: String { get }
+    var signer: Signer { get }
+    var idKind: IdentityKind { get }
+
+    // Hook used by AuthSchemeMiddleware that allows signing properties customization, if needed by an auth scheme
+    func customizeSigningProperties(signingProperties: Attributes, context: HttpContext) throws -> Attributes
+}
+
+extension AuthScheme {
+    func identityResolver(config: IdentityResolverConfiguration) -> (any IdentityResolver)? {
+        return config.getIdentityResolver(identityKind: self.idKind)
+    }
+}
```

Comment:
consider moving this function out of extension into the base definition

Reply:
If by base definition you mean the original protocol definition, Unfortunately it's not possible to add method body to a method in a Swift protocol.

Reply:
I meant should it be inside of an extension vs where you define AuthScheme

Reply:
oh wait nevermind all good